aten/src/ATen/native/cpu/Loops.h

```diff
@@ -208,6 +284,40 @@ static inline void vectorized_outer_reduction(char** data, int64_t inner_stride,
   });
 }
 
+template <typename func_t>
+void unary_kernel(TensorIterator& iter, func_t op) {
+  using traits = unary_function_traits<func_t>;
+
+  iter.for_each([&](int ntensor, char** data, const int64_t* strides, int64_t n) {
+    // Specializations to encourage auto-vectorization (trick from Numpy's loops.c.src)
+    if (is_unary_contiguous<traits>(strides)) {
+      unary_loop(data, strides, 0, n, op);
+    } else if (is_unary_contiguous_s1<traits>(strides)) {
+      unary_loop(data, strides, 0, n, op);
+    } else {
+      unary_loop(data, strides, 0, n, op);
```

Comment:
unary_loop(data, strides, 0, n, op); in all 3 branches?

Reply:
just like here: https://github.com/pytorch/pytorch/pull/18309/files/33dbaaa7c5974b52ee78862624c3c494f526c67f#diff-0d1178f1a4ce15aeb760d251974e6924R322

Reply:
We should leave comments in places like that!

Reply:
I'm writing up docs on TensorIterators along the way. Since this is a fix to an issue I won't add unnecessary doc roundtrips.